resource-aggregate/events/resourceStateSnapshotTaken.go

```diff
@@ -172,8 +173,28 @@ func (e *ResourceStateSnapshotTaken) ValidateSequence(eventMetadata *EventMetada
 	return false
 }
 
+func Equal(current, changed *ResourceChanged) bool {
+	if current.Status != changed.Status {
+		return false
+	}
+
+	if current.Content.CoapContentFormat != changed.Content.CoapContentFormat ||
+		current.Content.ContentType != changed.Content.ContentType ||
+		!bytes.Equal(current.Content.Data, changed.Content.Data) {
+		return false
+	}
+
+	if current.AuditContext.UserId != changed.AuditContext.UserId ||
+		current.AuditContext.CorrelationId != changed.AuditContext.CorrelationId {
+		return false
+	}
+
+	return true
+}
+
 func (e *ResourceStateSnapshotTaken) HandleEventResourceChanged(ctx context.Context, changed *ResourceChanged) (bool, error) {
-	if e.ValidateSequence(changed.GetEventMetadata()) {
+	if e.ValidateSequence(changed.GetEventMetadata()) &&
+		(e.LatestResourceChange != nil && !Equal(e.LatestResourceChange, changed)) {
```

Comment:
and what `if e.LatestResourceChange == nil` ?

Reply:
yes, this is incorrect, if `LatestResourceChange == nil` then we shouldn't skip the event, we just don't have the cache value set